### PR TITLE
Improve handling of padding in multi_cell - fix #1001

### LINKED ISSUE #1001: Padding interferes with YPos.TOP
Hi guys, another really minor issue I encountered in this very useful library related to `YPos.TOP` handling in `multi_cell`.

It seems to me that `fpdf.py:3499` is duplicated at `fpdf.py:3526` (lines on master) which leads to an accumulation of the padding amount if that feature is used.

Describe the bug

**Error details**
Padding is accumulated in y-position despite using new_y=TOP

**Minimal code**
Please include some minimal Python code reproducing your issue:
```python
from fpdf import FPDF, YPos
pdf = FPDF()
pdf.add_page()
pdf.set_font('helvetica', '', 10)
pdf.set_fill_color(235)
for ch in 'ABCD':
    pdf.multi_cell(w=30, text=ch, align='CENTER', padding=5, fill=True, border=True, new_y=YPos.TOP)
pdf.output('test.pdf')
```
image

Expected behaviour is all cells would be in a single row. Commenting out the duplicated line recovers the expected behaviour. Happy to make a PR but looks like a one-liner to me.

### PR DESCRIPTION
**Pull request by mjasperse**
Fixes #1001 by adjusting where padding is accumulated for different combinations of `new_x` and `new_y`, as well as explicitly removing padding between components cells of multiline strings to avoid issues with `fill=True`.

As mentioned in the ticket I did start on a wider refactoring of `multi_cell` to draw a single filled background instead of rendering each line independently, but that's substantially more involved and easily breaks test cases, so potentially it makes sense to consider these smaller changes first.

**Checklist**:

- [ ] The GitHub pipeline is OK (green), 
      meaning that both `pylint` (static code analyzer) and `black` (code formatter) are happy with the changes of this PR.

- [X] A unit test is covering the code added / modified by this PR

- [X] This PR is ready to be merged 

- [ ] In case of a new feature, docstrings have been added, with also some documentation in the `docs/` folder. *N/A*

- [X] A mention of the change is present in `CHANGELOG.md`

By submitting this pull request, I confirm that my contribution is made under the terms of the [GNU LGPL 3.0 license](https://github.com/py-pdf/fpdf2/blob/master/LICENSE).

### CODE REVIEW

File: fpdf/fpdf.py [hunk 1]
```diff
@@ -3511,22 +3512,23 @@ def multi_cell(
             clearance_margins.append(self.c_margin)
         if not padding.right:
             clearance_margins.append(self.c_margin)
-        self.x += padding.left
+        if align != Align.X:
```
Comment (gmischler) [inline fpdf/fpdf.py:3515]: Nice catch! 👍 

I see you have added tests for all alignment options together with padding, which we apparently didn't have before.

**Review by gmischler — APPROVED** [review]
Nice work, good tests!

**Comment by gmischler** [comment]
It seems that several tests are failing. Can you please check out why?

**Comment by Lucas-C** [comment]
Those are the 3 failing tests:
```
test/table/test_table_padding.py::test_multicell_with_padding
test/table/test_table_padding.py::test_multicell_return_value
test/table/test_table_padding.py::test_table_vertical_alignment
```

**Comment by mjasperse** [comment]
Hi guys, I looked into the failing tests and my interpretation is below. Although it is generally my last resort to change the expected outputs, I think potentially the new behaviour as described is preferable. If you agree I'm happy to regenerate the expected PDFs and update the PR. If you disagree please confirm the desired behaviour and I'll have another look at changing the handling.


**test_table_vertical_alignment**
It looks like the accumulation of padding in y was causing the layout to insert a pagebreak before the final row. It seems that the final row _can_ fit on the first page so I think it is desirable to put it there.

image

(New behaviour on the left, old behaviour on the right)


**test_multicell_with_padding** and **test_multicell_return_value**
This is related to the overlapping cell borders that are produced when the multicell contains multiple lines, since the padding is applied to the box around each line resulting in border segments that overlap. The new behaviour is that the borders should no longer overlap, causing the comparison to fail.

image

(New behaviour on the left, old behaviour with overlapping border on the right - easier to confirm the border lines by importing into inkscape or similar)

**Comment by gmischler** [comment]
> If you agree I'm happy to regenerate the expected PDFs and update the PR

Updating the reference PDFs is frequently inevitable, ideally because the internal structure has changed without affecting the visual appearance. But in your case the current files are actually suboptimal even in appearance, and the new ones show what we really want. So yes, please update them.

**Comment by gmischler** [comment]
All checks green, merging now.

Nice work, @mjasperse, thank you for the contribution!

**Comment by Lucas-C** [comment]
@allcontributors please add @mjasperse for code

**Pull request merged**